backend/fms_core/services/dataset.py

```diff
@@ -219,7 +238,187 @@ def get_experiment_run_lane_validation_status(experiment_run_id: int, lane: int)
 
     return validation_status, errors, warnings
 
+def set_dataset_release_status(dataset_id: int, readsets_release_status: dict[str, ReleaseStatus], released_by: User):
+    """
+    Set release_status for readsets of the given dataset.
+
+    Args:
+        `dataset_id`: Freezeman Dataset ID.
+        `readsets_release_status`: dictionary mapping readset id to a valid release status (choices : Available - 0 (default), Released - 1, Blocked - 2).
+        `released_by`: The user that set the release status.
+    
+    Returns:
+        A tuple of the count of release status set, errors and warnings
+    """
+    count_status = 0
+    released_count = 0
+    readsets_released = []
+    readsets_recalled = []
+    is_status_reset = False
+    dataset_obj = None
+    errors = []
+    warnings = []
+
+    if dataset_id is None:
+        errors.append(f"Missing dataset id.")
+    if any([id for id, status in readsets_release_status.items() if status not in ReleaseStatus.values]):
+        errors.append(f"The release status can only be {' or '.join([f'{value} ({name})' for value, name in ReleaseStatus.choices])}.")
+    if released_by is None or not isinstance(released_by, User):
+        errors.append(f"Missing released_by.")
+
+    if not errors:
+        try:
+            dataset_obj = Dataset.objects.get(id=dataset_id)
+        except Exception as e:
+            errors.append(f"Failed to get Dataset {dataset_id}.")
```

Comment:
Why go with exception if you can instead optionally get a Dataset object and append error if it's `None`?

Reply:
yeah i will return in case it cannot find a dataset.